[clkmgr/rstmgr/doc] Add description of LPG output signals (#18105)
lowRISC/opentitan: hw/ip/clkmgr/doc/theory_of_operation.md

* [Interface Tables](../../../top_earlgrey/ip/clkmgr/data/autogen/clkmgr.hjson#interfaces)

> ### Clock Gated Indications for Alert Handler

Reviewer: Move this above the Hardware Interfaces section?

Author: done

> The alert handler needs to know the status of the various clock domains in the system so that the ping mechanism does not produce false alert indications.

Author: done

> To that end, the clock manager outputs a 4bit MuBi signal for each clock domain that indicates whether it is active or not.

Reviewer: ```suggestion
To that end, the clock manager outputs a 4bit MuBi signal for each clock domain that indicates whether it is active.
```

Author: done

> The divided frequency is not assumed to be synchronous with its source and are thus treated like another asynchronous branch.

Reviewer: Huh, this section is misleading: I think line 186 should explain clock division is required for peripherals that need clocks slower than IO. In normal operation the IO clock is divided by 2 and 4.

Author: I agree - I have reworded this part, PTAL.

Further, the clock dividers are hardwired and have no software control, this is to further ensure there are no simple paths for faulty or malicious software to tamper.

> Clock division is mainly used as part of the [external clock switch support](#external-clock-switch-support), since the external clock may be slower than the internal clock source.

Reviewer: ```suggestion
Clock division can be modified for [external clock switch support](#external-clock-switch-support), since the external clock may be slower than the internal clock source.
```

Author: done
